FFmpegInterop/Source/UncompressedVideoSampleProvider.cpp

```diff
@@ -27,18 +27,29 @@ extern "C"
 
 
 using namespace FFmpegInterop;
+using namespace Windows::Media::MediaProperties;
 
 UncompressedVideoSampleProvider::UncompressedVideoSampleProvider(
 	FFmpegReader^ reader,
 	AVFormatContext* avFormatCtx,
 	AVCodecContext* avCodecCtx)
 	: UncompressedSampleProvider(reader, avFormatCtx, avCodecCtx)
-	, m_pSwsCtx(nullptr)
 {
-	for (int i = 0; i < 4; i++)
+	switch (m_pAvCodecCtx->pix_fmt)
 	{
-		m_rgVideoBufferLineSize[i] = 0;
-		m_rgVideoBufferData[i] = nullptr;
+	case AV_PIX_FMT_YUV420P:
+	case AV_PIX_FMT_YUVJ420P:
+		m_OutputPixelFormat = AV_PIX_FMT_YUV420P;
+		OutputMediaSubtype = MediaEncodingSubtypes::Iyuv;
+		break;
+	case AV_PIX_FMT_YUVA420P:
+		m_OutputPixelFormat = AV_PIX_FMT_BGRA;
```

Comment:
Isn't there a typo on AV_PIX_FMT_BGRA ?

Reply:
Nope, ffmpeg BGRA = MF ARGB

Reply:
Ok, then it is AVPixelFormat::AV_PIX_FMT_RGBA in WriteAVPacketToStream that's wrong no ?

Reply:
Fixed, thanks. It still worked (by chance) because LineSize[1] is 0 for BGRA...